app/src/main/java/org/piwigo/ui/login/LoginActivity.java

```diff
@@ -140,12 +141,14 @@ public void finish() {
             }
             authenticatorResponse = null;
         }
+
         super.finish();
     }
 
     private void loginSuccess(LoginResponse response) {
         fabProgressCircle.hide();
         if (viewModel.isEditExisting()) {
+            startActivity(new Intent(getApplicationContext(), MainActivity.class));
```

Comment:
doesn't this start the MainActivity again on top of the backstack instead of navigating back to the MainActivity instance already there? I believe we could add onActivityResult in ManageAccountsActivity and call finish there, also to fully rewind the stack. Or start MainActivity as a new task? e.g. by something like

`Intent intent = new Intent(getApplicationContext(), MainActivity.class);
intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK);
startActivity(intent);`

Reply:
I actually thought about it and I think I expect that I end up in the account manager if I press back after adding an account lands me on the main view. But yes, if I come here for the first time, I don't want 2 main views. I'll extend the test to your user story

Reply:
the main view 'eats' all back presses - there is no stack processing

Reply:
let's take that to #225 and come back to automated tests for it afterwards